packages/core/src/utils/is-empty.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Checks if the given value is empty.
+ *
+ * @param {unknown} value - The value to check.
+ * @returns {boolean} Returns `true` if the value is empty, otherwise `false`.
+ *
+ * @example
+ * isEmpty(null); // => true
+ * isEmpty(''); // => true
+ * isEmpty([]); // => true
+ * isEmpty({}); // => true
+ * isEmpty('Hello'); // => false
+ * isEmpty([1, 2, 3]); // => false
+ * isEmpty({ a: 1 }); // => false
+ */
+const isEmpty = (value: unknown): boolean => {
```

Comment:
Lets use https://www.npmjs.com/package/lodash.isempty instead.
Can send this in a followup PR if needed.

Reply:
Wouldn't that be good to reduce the dependencies?

Reply:
updated with 48dc84f883bdf4d42f3739dcad2c0bf4cd3691b3